Civi/Token/TokenRow.php

```diff
@@ -183,6 +183,9 @@ public function tokens($a = NULL, $b = NULL, $c = NULL) {
    */
   public function customToken($entity, $customFieldID, $entityID) {
     $customFieldName = 'custom_' . $customFieldID;
+    if (empty($entityID)) {
+      return $this->format('text/html')->tokens($entity, $customFieldName, '');
+    }
     $record = civicrm_api3($entity, 'getSingle', [
```

Comment:
Can we stop this API call from throwing an exception when rendering a token? That's a hard crash. Seems like an anti-pattern we would want to avoid.

Reply:
yeah - we could add a try-catch on the getsingle - that doesn't void this though as we would also want to not to the extra call